fem/nonlinearform_ext.hpp

```diff
@@ -17,28 +17,75 @@
 
 namespace mfem
 {
+
 class NonlinearForm;
+class NonlinearFormIntegrator;
 
+/** @brief Class extending the NonlinearForm class to support the different
+    AssemblyLevel%s. */
 class NonlinearFormExtension : public Operator
 {
 protected:
-   NonlinearForm *n; ///< Not owned
+   const NonlinearForm *nlf; ///< Not owned
+
 public:
-   NonlinearFormExtension(NonlinearForm *form);
-   virtual void AssemblePA() = 0;
+   NonlinearFormExtension(const NonlinearForm*);
+
+   /// Assemble at the AssemblyLevel of the subclass.
+   virtual void Assemble() = 0;
```

Comment:
This would technically be a breaking change if introduced. I'm don't think it'll affect anyone, since this is largely used internally within MFEM, but it is something we should be mindful about.

Reply:
Discussed in a MFEM meeting, and it was decided that this was an appropriate change due it correcting the previous name which incorrectly suggested that the method should only be used for partial assembly rather than for PA, EA, or FA.